src/index.ts

```diff
@@ -108,6 +117,7 @@ export default class HDKeyring implements Keyring<SerializedHDKeyring> {
       keyringType: HDKeyring.type,
       path: this.path,
       addressIndex: this.#addressIndex,
+      passphrase: this.#passphrase,
```

Comment:
@mhluongo forgot the one thing I wanted to ask before merging: what’s your preference on offering a secure default that doesn’t serialize pass phrase vs defaulting to serializing it?

Reply:
I don't use BIP-39 wallets regularly, so I'm not sure what the state of play is / what user expectations are these days. Generally I'd rather not store a passphrase if we can avoid it :thinking: 

Not doing it has some interesting implications for [`extension`](https://github.com/tallycash/extension/pull/1022) though. In particular, do BIP-39 wallets need to ask for additional passphrases over what we use for the wallet? Or is the idea that they can be the same passphrase?

Reply:
Tl;dr: After digging a bit more on passphrases, why they exist, and their intended use and security model, I have a stronger belief that hd-keyring should not store them by default. In fact, I'm tempted to say it shouldn't support storing passphrases as a first-class concept at all, instead accepting them as arguments at points where they would be necessary. I also think supporting passphrases properly in the UI will require reframing keyrings with passphrases as conceptually more similar to an external signer similar to a hardware wallet. More below.

> If we consider this place 'safe enough' to store the mnemonic, why not the data that gets combined with it to derive addresses?

Unless I'm missing something, the answer here is straightforward: if it were equally secure to store mnemonic and passphrase alongside each other, why would we have a passphrase at all? Presumably the idea is that the private key material is stored in one way, and the passphrase in a different, orthogonal way. If they're stored alongside each other the passphrase gives no additional security, right?

See [this guide from Trezor](https://blog.trezor.io/passphrase-the-ultimate-protection-for-your-accounts-3a311990925b#d30c) on what the security benefits are, and [this note from the Bitcoin wiki](https://en.bitcoin.it/wiki/Seed_phrase#Two-factor_seed_phrases) on two-factor seed phrases.

> Because the extension uses deserialize when populating keyrings in unlock(), the passphrase would need to be provided there if it was not stored here because AddAddressesSync is called. Additional refactoring would likely be required on both sides.

This feels rough, but points to an overall complication with supporting these directly in the first place. They feel like they'll have an interaction more like an external signer (where additional interactions may need to occur before derivation, signing, etc, such as plugging in a USB device for a hardware wallet, and we need to remember more in the extension). Thinking of entering the passphrase as a step similar to plugging in a hardware wallet feels like a good mental model, and reflects the second-factor nature of passphrases IMO.

This should be somewhat straightforward to support once tallycash/extension#1165 is implemented, since it'll just be another signer type.

> I don't want to have to enter these every single time that I need to use them, but want to have an option to require it in the future (so users that want the extra security can have it)

Last note: this feels implementable with some levers in the extension.

Reply:
@Shadowfiend,
  
Thank you for the detailed and thoughtful insight!
  
I read the RFB in the PR you linked and really like the separation of duties there. That effort will definitely help with future additions to the wallet and addresses technical debt early.
  
I pushed changes to remove the storage of the passphrase. Let me know if it looks good now